src/helpers/job.coffee

```diff
@@ -66,12 +67,16 @@ JobListenerConfig = (checkIntervals, checkRepeats) ->
       # whenever this job status is updated, emit the status for this wrapper
       updateJobStatus = @_updateJobStatusFor(id)
       JobStore.on("job.#{jobId}.*", updateJobStatus)
-      JobStore.off("job.#{jobId}.final", updateJobStatus)
+      JobStore.once("job.#{jobId}.final", @stopListening(updateJobStatus))
 
     stopChecking: (id) ->
       jobId = @_getLatestJob(id)
       JobActions.stopChecking(jobId) if jobId?
 
+    stopListening: (updateJobStatus) ->
+      (jobData) ->
+        JobStore.off("job.#{jobData.id}.*", updateJobStatus)
+
```

Comment:
I see what you're doing here, but it took me a bit to realize that `stopListening` is a factory method that returns a function which stops binding the updateJobStatus when it's called.

Maybe we could call it `buildStopListeningHelper` ?  Not super happy with that either, but maybe you can come up with something better, `stopListening` just sounds like it's going to immediately stop listening, not return a function that will eventually do so.